File: openFrameworks_testing/src/expansion_api.cpp
//
//  expansion_api.cpp
//  openFrameworks_testing
//
//  Created by Kevin Nottberg on 2/9/18.
//

#include "expansion_api.hpp"

int hypoRadiusCorner(Bubble &selectedBubble) {
    float xLength = 0;
    float yLength = 0;
    
    if(selectedBubble.position.x_comp > (ofGetWindowWidth() / 2)) {
        xLength = selectedBubble.position.x_comp;
    } else {
        xLength = (ofGetWindowWidth() - selectedBubble.position.x_comp);
    }
    
    if(selectedBubble.position.y_comp > (ofGetWindowHeight() / 2)) {
        yLength = selectedBubble.position.y_comp;
    } else {
        yLength = (ofGetWindowHeight() - selectedBubble.position.y_comp);
    }
    
    return int(sqrt((xLength * xLength) + (yLength * yLength)));
}

void calculateSetExpansiveDeltas(Bubble &selectedBubble, vector<Bubble> &bubbles, int expansionRate) {
    selectedBubble.setExpansiveVelocity(0, 0);
    
    for(int i = 0; i < bubbles.size(); i += 1) {
        bubbles[i].preExpansivePos.x_comp = bubbles[i].position.x_comp;
        bubbles[i].preExpansivePos.y_comp = bubbles[i].position.y_comp;
        
        if(bubbles[i].ID != selectedBubble.ID) {
            float deltaX = (selectedBubble.position.x_comp - bubbles[i].position.x_comp) * -1;
            float deltaY = (selectedBubble.position.y_comp - bubbles[i].position.y_comp) * -1;
            
            bubbles[i].setExpansiveVelocity(deltaX, deltaY);
            bubbles[i].expansiveVelocity.normalize();
            bubbles[i].expansiveVelocity.multi(expansionRate);
        }
    }
}

void reverseExpansiveDeltas(vector<Bubble> &bubbles) {
    for(int i = 0; i < bubbles.size(); i += 1) {
        bubbles[i].expansiveVelocity.x_comp *= -1;
        bubbles[i].expansiveVelocity.y_comp *= -1;
    }
}

void expand(Bubble &selectedBubble, int expansionRate) {
    selectedBubble.radius += expansionRate;
    cout<< selectedBubble.radius << endl;
}

void close(Bubble &selectedBubble, int expansionRate) {
    selectedBubble.radius -= expansionRate;
}

File: Floating_Note_OpenGL/ExpansionAPI.py
import math, datetime
from simple_vector_library import *

def hypoRadiusCorner(selectedObj, xBound, yBound):
    xLength = 0
    yLength = 0

    if(selectedObj.pos.x_component > (xBound/2)):
        xLength = selectedObj.pos.x_component
    if(selectedObj.pos.x_component < (xBound/2)):
        xLength = (xBound - selectedObj.pos.x_component)

    if(selectedObj.pos.y_component > (yBound /2)):
        yLength = selectedObj.pos.y_component
    if(selectedObj.pos.y_component < (yBound / 2)):
        yLength = (yBound - selectedObj.pos.y_component)

    return int(math.pow(math.pow(xLength, 2) + math.pow(yLength, 2), .5))


def calculateExpansiveDeltas(objs, selectedObj, expansionRate):
    selectedObj.expansiveVelcoity = Py_Vector(0, 0)

    for obj in objs:
        obj.preExpansivePos = Py_Vector(obj.pos.x_component, obj.pos.y_component)
        if obj.ID != selectedObj.ID:
            deltaX = ((selectedObj.pos.x_component - obj.pos.x_component) * -1)
            deltaY = ((selectedObj.pos.y_component - obj.pos.y_component) * -1)

            obj.expansiveVelocity = Py_Vector(deltaX, deltaY)
            obj.expansiveVelocity.normalize()
            obj.expansiveVelocity.multi(expansionRate)

def reverseExpansiveDeltas(objs):
    for obj in objs:
        obj.expansiveVelocity.x_component *= -1
        obj.expansiveVelocity.y_component *= -1

def expand(selectedObj):
    selectedObj.radius += 30

def close(selectedObj):
    selectedObj.radius -= 30
